django/db/models/sql/query.py

```diff
@@ -2014,11 +2019,15 @@ def split_exclude(self, filter_expr, can_reuse, names_with_path):
             lookup = lookup_class(pk.get_col(query.select[0].alias), pk.get_col(alias))
             query.where.add(lookup, AND)
             query.external_aliases[alias] = True
+        else:
+            lookup_class = select_field.get_lookup("exact")
+            lookup = lookup_class(col, ResolvedOuterRef(trimmed_prefix))
+            query.where.add(lookup, AND)
 
-        lookup_class = select_field.get_lookup("exact")
-        lookup = lookup_class(col, ResolvedOuterRef(trimmed_prefix))
-        query.where.add(lookup, AND)
-        condition, needed_inner = self.build_filter(Exists(query))
+        exists = Exists(query)
+        if negate:
+            exists = ~Q(exists)
```

Comment:
Does this _need_ to be wrapped into a `Q` (i.e. for API reasons that I can't "_see_" in the diff)? Off the top of my head, I _feel_ like `~Exists(query)` ... exists (though I may be mis-remembering) and could be something like:

```suggestion
        if negate:
            exists = ~exists
```

Reply:
I explicitly used `~Q(exists)` because the last commit is able to reduce `~Q(~Q(foo)))` to `foo` while `~Exists` is turned into `~Q(Exact(NegatedExpression(Exists(...)), True))` which is not reducable and thus generates SQL that is harder to interpret when debugging this problem.